### DISCUSSION TITLE
Using dynamic styles and CSP

### DISCUSSION THREAD

**Discussion by BMCwebdev** [facebook/stylex#475] [Q&A] [answered]
Morning. Oh wait, it's 1:00.

Does anyone have any advice on how to implement dynamic styles with StyleX when you also employ a Content Security Policy that enforces no inline CSS? How could we hash or nonce it, and when in the build or run would this happen? Thanks!

**Comment by nmn** [comment, marked as answer]
Dynamic styles are well, ... dynamic. So they are generated at runtime with JS and applied as inline styles. They can't be hashed because they aren't known ahead of time.

That said, in the VAST, VAST majority of cases you don't need dynamic styles at all. So, perhaps look into alternatives to avoid using dynamic styles and use more expressive static CSS instead?

Some very interesting things can be achieved with CSS and Math.

**Reply by BMCwebdev** [reply]
Thanks @nmn I haven’t actually run into the need for it yet, and I’ve never previously had to deal with a CSP that disallowed it (but that might be headed our way), so I’ve never hashed or nonce-ed(?) CSS in my career. Thought I’d ask ahead of time.
